src/python/pants/task/target_restriction_mixins.py

```diff
@@ -0,0 +1,67 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.task.goal_options_mixin import GoalOptionsRegistrar
+
+
+class HasTransitiveOptionMixin(object):
+  """A mixin for tasks that have a --transitive option.
+
+  Some tasks must always act on the entire dependency closure. E.g., when compiling, one must
+  compile all of a target's dependencies before compiling that target.
+
+  Other tasks must always act only on the target roots (the targets explicitly specified by the
+  user on the command line). E.g., when finding paths between two user-specified targets.
+
+  Still other tasks may optionally act on either the target roots or the entire closure,
+  as the user prefers in each case. E.g., when invoking a linter. This mixin supports such tasks.
+
+  Note that this mixin doesn't actually register the --transitive option. It assumes that this
+  option was registered on the task (either directly or recursively from its goal).
+  """
+
+  def get_targets(self, predicate=None):
```

Comment:
Because this method doesn't call `super`, the declaration order of mixins will matter. Should be made explicit if it is intentional. I'm biased in general toward only overriding abstract methods, so this feels pretty fidgety.

Reply:
It is a little fiddly. But useful... I mean, we want the order to matter, as long as it's correctly constructed. Order mattering is pretty common with mixins, I think.